### PR TITLE
refactor: provider

### PR DESCRIPTION
**Pull request by mikelxk**
This commit put metadata of provider within the provider itself.

### CODE REVIEW

File: src/composition/auth.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ const openLogin = (source: string) => {
   return window.open(url, '_blank');
 };
 
-const loginSourceList = MediaService.getLoginProviders().map((i) => i.name);
+const loginSourceList = MediaService.getLoginProviders().map((i) => i._name);
```
Comment (listen1) [inline src/composition/auth.ts:27]: if name can be accessed outside, it's better to rename it to name without underscore

**Comment by listen1** [comment]
it's really nice to move music provider properties back to their own file, more clear way to manage those data.

File: src/composition/auth.ts (hunk 1 above)
Comment (mikelxk) [reply src/composition/auth.ts:27]: Static property 'name' conflicts with built-in property 'Function.name'

Comment (listen1) [reply src/composition/auth.ts:27]: maybe add `getName()` with `_name` property?

Comment (mikelxk) [reply src/composition/auth.ts:27]: if _name is reserved for internal use. What about `Name`?

**Pull request merged**